### PR TITLE
Add new requirement notice fix #57

### LINKED ISSUE #57: Add new requirement notice
"Gitium requires minimum PHP 5.3 version"

### PR DESCRIPTION
**Pull request by Marius786**

### CODE REVIEW

File: gitium/inc/class-gitium-requirements.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+	private function is_php_version_gt_5_3() {
+		if ( ! function_exists( 'phpversion' ) ) {
+			return array( false, 'The function `phpversion` is disabled!' );
+		} else {
+			$php_version = phpversion();
+			if ( '5.3' <= substr( $php_version, 0, 3 ) ) {
+				return array( true, "The PHP version is `$php_version`." );
+			} else {
+				return array( false, "The PHP version is `$php_version` and is not greater/equal than/with 5.3!" );
```
Comment (bogdanpetrea) [inline gitium/inc/class-gitium-requirements.php]: "The PHP version is `$php_version` and is not greater or equal to 5.3!" sounds better I think.

File: gitium/inc/class-gitium-requirements.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+	private function is_proc_open() {
+		if ( ! function_exists( 'proc_open' ) ) {
+			return array( false, 'The function `proc_open` is disabled!' );
+		} else {
+			return array( true, 'The function `proc_open` is enabled!' );
+		}
+	}
+
+	private function is_php_version_gt_5_3() {
```
Comment (bogdanpetrea) [inline gitium/inc/class-gitium-requirements.php]: Maybe we should have the minimum PHP version defined as a constant in one place, so when we have to bump the version we just change that constant.

File: gitium/inc/class-gitium-submenu-requirements.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+	private function is_proc_open() {
+		if ( ! function_exists( 'proc_open' ) ) {
+			return array( false, 'The function `proc_open` is disabled!' );
+		} else {
+			return array( true, 'The function `proc_open` is enabled!' );
+		}
+	}
+
+	private function is_php_version_gt_5_3() {
```
Comment (bogdanpetrea) [inline gitium/inc/class-gitium-submenu-requirements.php]: This function is defined in class-gitium-requirements.php as well. What's the reason behind this?

File: gitium/inc/class-gitium-submenu-requirements.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+		$git_version = $git->get_version();
+
+		if ( '1.7' <= substr( $git_version, 0, 3 ) ) {
+			return array( true, "The git version is `$git_version`." );
+		} else {
+			return array( false, "The git version is `$git_version` and must be greater than `1.7`!" );
+		}
+	}
+
+	private function is_proc_open() {
```
Comment (bogdanpetrea) [inline gitium/inc/class-gitium-submenu-requirements.php]: This function is defined in class-gitium-requirements.php class Gitium_Requirements as well. What's the reason behind this?

File: gitium/inc/class-gitium-submenu-requirements.php (hunk 3 above)
Comment (Marius786) [inline gitium/inc/class-gitium-submenu-requirements.php]: The gitium-submenu-requirements.php is replaced by class-gitium-requirements.php

File: tests/test-gitium.php [hunk 5]
```diff
@@ -67,10 +67,6 @@ function test_has_action_gitium_hook_plugin_and_theme_editor_page() {
 		$this->assertGreaterThan( 0, has_action( 'admin_enqueue_scripts','gitium_hook_plugin_and_theme_editor_page' ) );
 	}
 
-	function test_has_action_gitium_require_minimum_version() {
```
Comment (bogdanpetrea) [inline tests/test-gitium.php]: Maybe write some tests for the new functionalities.

**Pull request merged**